### PR TITLE
test: add utility for namespace labels patching and add the test cases for denyNodeFeatureLabels restriction

### LINKED ISSUE #1879: Improve CR Restrictions e2e tests
**What would you like to be added**:
- Test the case where we have a non 3rd party NodeFeature and the `disableFeatureLabels` restriction
- Add utility to patch namespaces

**Why is this needed**:
- Cover more edge cases and improve confidence in the feature. Hopefully, we can remove the experimental flag.

### PR DESCRIPTION
**Pull request by TessaIO**
Resolves #1879

### CODE REVIEW

**Comment by TessaIO** [comment]
/cc @marquiz 
Maybe we can add this for v0.17 also

File: test/e2e/utils/namespace/namespace.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	"sigs.k8s.io/node-feature-discovery/pkg/utils"
+)
+
+const (
+	AddOperation    = "add"
+	RemoveOperation = "remove"
```
Comment (marquiz) [inline test/e2e/utils/namespace/namespace.go:33]: I think it would make sense to specify these in `pkg/utils/jsonpatch.go`

Again, this could be a separate PR converting all uses of `utils.NewJsonPatch` to use these definitions.

WDYT?

Also, bike-shedding on the naming, for the generic pkg/utils version:

```suggestion
	JSONPatchOpAdd    = "add"
	JSONPatchOpRemove = "remove"
```

Comment (marquiz) [reply test/e2e/utils/namespace/namespace.go:33]: ping @TessaIO

Comment (TessaIO) [reply test/e2e/utils/namespace/namespace.go:33]: I'll change this ASAP

**Review by marquiz — APPROVED** [review]
Thank you @TessaIO for making this improvement. Looks good to me now
/lgtm
